hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/client/common/HoodieFlinkEngineContext.java

```diff
@@ -86,6 +90,17 @@ public RuntimeContext getRuntimeContext() {
         .collect(Collectors.toList());
   }
 
+  @Override
+  public <I, K, V> Stream<ImmutablePair<K, V>> mapPartitionsToPairAndReduceByKey(
+      Stream<I> data, SerializablePairFlatMapFunction<Iterator<I>, K, V> flatMapToPairFunc,
+      SerializableBiFunction<V, V, V> reduceFunc, int parallelism) {
+    return throwingFlatMapToPairWrapper(flatMapToPairFunc).apply(data.parallel().iterator())
```

Comment:
we can't leverage "parallelism" value in flink transformation functions is it ?

Reply:
We cannot.  The `parallelism` is introduced to be compatible with Spark.  All other Flink transformations don't use the parallelism argument either.